app/components/PlacePopup.tsx

```diff
@@ -52,6 +54,7 @@ const PlacePopup = ({ map, place, show, onClose }: PopupProps) => {
 
   useEffect(() => {
     if (show && map) {
+      console.log("Showing Popup for:", place);
       popupRef.current?.addTo(map);
       map.flyTo({ center: coordinates.current, zoom: 15 });
```

Comment:
You might want to add an optional `zoomToFeature` boolean property. Zooming to the feature during search updates the results.